pkg/daemon/daemon.go

```diff
@@ -429,13 +429,19 @@ func (dn *Daemon) nodeStateSyncHandler() error {
 				log.Log.Error(err, "nodeStateSyncHandler(): failed to check if sriov-config service exist on host")
 				return err
 			}
+			postNetworkServiceEnabled, err := dn.HostHelpers.IsServiceEnabled(systemd.SriovPostNetworkServicePath)
+			if err != nil {
+				log.Log.Error(err, "nodeStateSyncHandler(): failed to check if sriov-config-post-network service exist on host")
+				return err
+			}
 
 			// if the service doesn't exist we should continue to let the k8s plugin to create the service files
 			// this is only for k8s base environments, for openshift the sriov-operator creates a machine config to will apply
 			// the system service and reboot the node the config-daemon doesn't need to do anything.
-			if !serviceEnabled {
+			if !(serviceEnabled && postNetworkServiceEnabled) {
```

Comment:
This systemd handling part of the daemon starting to get long, we should split it to separate "handleSystemdMode" method.

that said, its not related to this PR....

Reply:
yes, in follow up PRs I will move things a bit in the function.